### PR TITLE
Finished refactored

### PR DESCRIPTION
**Pull request by karki011**
I finished refactored and fixed the bug. Please let me your thoughts.

### CODE REVIEW

File: CommonErrors.Shared/CommonAnswer.cs [hunk 1]
```diff
@@ -2,6 +2,8 @@
 {
     public class CommonAnswer : IGradable
     {
-        public decimal Grade { get; set; }
+        private decimal grade;
+        public decimal Grade => grade;
+        public void SetGrade(decimal value) => grade = value;
```
Comment (codytwinton) [inline CommonErrors.Shared/CommonAnswer.cs]: This refactor added 2 lines of code, which is unnecessary. The previous implementation was the best.

File: CommonErrorsKata/QuizForm.cs [hunk 2]
```diff
@@ -13,67 +13,81 @@ public partial class CommonErrorsForm : Form
...
+        private readonly string[] possibleAnswers;
+        bool IsPlaying = true;
 
         public CommonErrorsForm()
         {
             InitializeComponent();
             synchronizationContext = SynchronizationContext.Current;
-            files = System.IO.Directory.GetFiles(Environment.CurrentDirectory +  @"..\..\..\ErrorPics");
-            possibleAnswers = new string[] { "Missing File", "null instance", "divide by zero" };
+            files = System.IO.Directory.GetFiles(Environment.CurrentDirectory + @"..\..\..\ErrorPics");
```
Comment (codytwinton) [inline CommonErrorsKata/QuizForm.cs]: Neither my photo nor logo.png are stack trace errors. Delete those photos.

File: CommonErrorsKata/QuizForm.cs [hunk 3]
```diff
@@ -13,67 +13,81 @@ public partial class CommonErrorsForm : Form
...
+            possibleAnswers = files.Select(f => Path.GetFileNameWithoutExtension(f)).ToArray();
```
Comment (codytwinton) [inline CommonErrorsKata/QuizForm.cs]: Excellent implementation

File: CommonErrorsKata/QuizForm.cs [hunk 4]
```diff
@@ -13,67 +13,81 @@ public partial class CommonErrorsForm : Form
...
                 {
-                    UpdateProgress(i);
+                    UpdateProgress(_time);
                     Thread.Sleep(50);
                 }
-                Message("Need to be quicker on your feet next time!  Try again...");
+
+                if (IsPlaying == false)
+                {
+                    Application.Exit();
```
Comment (codytwinton) [inline CommonErrorsKata/QuizForm.cs]: return in this scope and the else scope will no longer be necessary.

File: CommonErrorsKata/QuizForm.cs [hunk 5]
```diff
@@ -13,67 +13,81 @@ public partial class CommonErrorsForm : Form
...
 
         private void LstAnswers_Click(object sender, EventArgs e)
         {
-            i = 100;
-            var tokens = currentBaseName.Split(' ');
-            //TODO:  Figure out what is a valid answer.
-            answerQueue.Enqueue(new TrueFalseAnswer(true));
+            _time = 100;
+            var selected = possibleAnswers[lstAnswers.SelectedIndex];
+            answerQueue.Enqueue(selected == currentBaseName ? new TrueFalseAnswer(true) : new TrueFalseAnswer(false));
```
Comment (codytwinton) [inline CommonErrorsKata/QuizForm.cs]: Replace this:

```cs
answerQueue.Enqueue(selected == currentBaseName ? new TrueFalseAnswer(true) : new TrueFalseAnswer(false));
```

with this:

```cs
answerQueue.Enqueue(new TrueFalseAnswer(selected == currentBaseName));
```

Simpler.

File: CommonErrorsKata/QuizForm.cs [hunk 6]
```diff
@@ -13,67 +13,81 @@ public partial class CommonErrorsForm : Form
...
+                IsPlaying = false;
                 MessageBox.Show("Congratulations you've defeated me!");
                 Application.Exit();
                 return;
             }
+
             label1.Text = answerQueue.Grade.ToString() + "%";
             var file = files.GetRandom();
-            currentBaseName= Path.GetFileName(file);
+            currentBaseName = Path.GetFileName(file)?.Replace(".png", "");
```
Comment (codytwinton) [inline CommonErrorsKata/QuizForm.cs]: You should probably use Path.GetFileNameWithoutExtension here as well.

File: CommonErrors.Shared/CommonAnswer.cs (hunk 1 above)
Comment (karki011) [reply CommonErrors.Shared/CommonAnswer.cs]: I have pushed the changes you pointed out.
Thank you for your feedback.

File: CommonErrorsKata/QuizForm.cs (hunk 2 above)
Comment (karki011) [reply CommonErrorsKata/QuizForm.cs]: I have pushed the changes you pointed out.
Thank you for your feedback.

File: CommonErrorsKata/QuizForm.cs (hunk 3 above)
Comment (karki011) [reply CommonErrorsKata/QuizForm.cs]: I have pushed the changes you pointed out.
Thank you for your feedback.

File: CommonErrorsKata/QuizForm.cs (hunk 4 above)
Comment (karki011) [reply CommonErrorsKata/QuizForm.cs]: I have pushed the changes you pointed out.
Thank you for your feedback.

File: CommonErrorsKata/QuizForm.cs (hunk 5 above)
Comment (karki011) [reply CommonErrorsKata/QuizForm.cs]: I have pushed the changes you pointed out.
Thank you for your feedback.

File: CommonErrorsKata/QuizForm.cs (hunk 6 above)
Comment (karki011) [reply CommonErrorsKata/QuizForm.cs]: I have pushed the changes you pointed out.
Thank you for your feedback.

**Pull request closed without merging**